### ISSUE TITLE
Nesting arrays when use XML2Array from rafrsr/lib-array2xml package

### ISSUE THREAD

**Issue by rafrsr** [rafrsr/lib-array2object#1]
When use `XML2Array::createArray()` from package [rafrsr/lib-array2xml](https://github.com/rafrsr/lib-array2xml)  with the following xml

``` xml
<entries>
    <entry>
        <name>someName</name>
    </entry>
    <entry>
        <name>someName2</name>
    </entry>
</entries>
```

return array like 

``` php
[entries] => Array
(
     [entry] => Array
         (
             [0] => Array
                 (
                     [name] => someName
                 )
             [1] => Array
                 (
                     [name] => someName2
                 )
             )
)
```

The following property not resolve this array correcly

``` php
/** @var array|Entry[] **/
protected $entries = [];
```

**Comment by rafrsr** [comment]
The same case but only with one children

``` xml
<entries>
    <entry>
        <name>someName</name>
    </entry>
</entries>
```

return array like

``` php
[entries] => Array
(
     [entry] => Array
         (
              [name] => someName
           )
)
```

doest not work with current solution ef9d8322c4983efc672a7c47e8e5da2d820cb182

**Comment by rafrsr** [comment]
fixed, commit d97a3c513593918290ab1b1083c9dc603aef2dc5
Now can use specific anotation

``` php
    /**
     * @var array|Object[]|[]
     */
    protected $entries;
```

> Note the `[]` after the object array, is used to ignore keys when create array of objects
